sdk/Pulumi/Config.cs

```diff
@@ -177,6 +186,20 @@ private T GetObjectImpl<T>(string key, string? use = null, [CallerMemberName] st
         public T GetObject<T>(string key)
             => GetObjectImpl<T>(key, nameof(GetSecretObject));
 
+        /// <summary>
+        /// Loads an optional configuration value, as an object, by its key, marking it as a secret
+        /// or null if it doesn't exist. This works by taking the value associated with <paramref
+        /// name="key"/> and passing it to <see cref="JsonSerializer.Deserialize{TValue}(string, JsonSerializerOptions)"/>.
+        /// </summary>
+        public Output<T>? GetSecretObject<T>(string key, JsonSerializerOptions options)
+        {
+            var v = GetImpl(key);
+            if (v == null)
+                return null;
+
+            return Output.CreateSecret(GetObjectImpl<T>(key, options: options)!);
```

Comment:
This retrieves the value twice. Once from GetImpl and once with GetObjectImpl. 

```
var v = GetObjectImpl<T>(key, options: options);
if (v == null)
  return null;
return Output.CreateSecret(v);
```

This is missing a `var v = GetObjectImpl<T>(key, options: options);
if (v == null)
  return null;
 return Output.CreateSecret(v);`

Reply:
Fixed